test/demo/utils/ofPuppeteer.ts

```diff
@@ -115,6 +127,20 @@ export class OFPuppeteerBrowser {
         return identity;
     }
 
+    /**
+     * Execute a function on the provider and return the result.
+     * @param fn Function to execute.
+     * @param args Arguments to provided to the function.
+     */
+    public async executeOnProvider<A extends any[], R, C extends ProviderWindow = ProviderWindow>(fn: (this: C, ...args: A) => R, ...args: A):
```

Comment:
As discussed, can we keep these specific utils separate from the `OFPuppeteer` parts to make it easier to extract down the line.

Reply:
Pulled out the stuff from this file, just changed the context generic so its not specific for client windows.

Reply:
Ok, ideally I would want to align this with the newer `OSPuppeter` stuff we have in notifications, but that can probably wait for now.